### PR TITLE
WIP: Authorization token forwarding.

### PR DESCRIPTION
**Pull request by alanconway**
Forwards Authorization header to all stores.

### CODE REVIEW

File: cmd/korrel8r/main.go [hunk 1]
```diff
@@ -88,6 +91,14 @@ func main() {
 		}
 		os.Exit(0)
 	}()
+	// Extra settings for REST configuration used by most stores and k8s clients.
+	k8s.ConfigureConfig = func(cfg *rest.Config) {
+		// TODO make these configurable. Use viper?
+		cfg.QPS = 100
+		cfg.Burst = 1000
+		cfg.Timeout = 5 * time.Second
+		cfg.Wrap(auth.Wrap)
```
Comment (xperimental) [inline cmd/korrel8r/main.go:100]: I think there's no need to go the roundtripper-route for the authentication. The `rest.Config` struct has an attribute for setting the bearer-token to use (`BearerToken`) along with a helper function to remove all other authentication options (`rest.AnonymousClientConfig()`):

```
cfg = rest.AnonymousClientConfig(cfg)
cfg.BearerToken = token
```

Note: The `token` here is the bare token, not the whole `Authorization` header.

File: pkg/domains/k8s/client.go [hunk 2]
```diff
@@ -52,8 +51,11 @@ func GetConfig() (*rest.Config, error) {
...
-	cfg.Burst = 1000
-	cfg.Timeout = 5 * time.Second
+	if ConfigureConfig != nil {
+		ConfigureConfig(cfg)
+	}
 	return cfg, nil
 }
+
+// ConfigureConfig if non-nil adds global settings to the common rest configuration.
+var ConfigureConfig func(cfg *rest.Config)
```
Comment (xperimental) [inline pkg/domains/k8s/client.go:61]: This is not really related to the authentication bits of this PR, but I dislike having a global variable for this purpose. I know this sounds like Java-EE, but I'd probably pass in a function that generates the client for me instead of calling a package-level function that relies on a global variable. This should also make testing more stable and parallelizable.

File: cmd/korrel8r/main.go (hunk 1 above)
Comment (alanconway) [reply cmd/korrel8r/main.go]: Injecting a roundtripper into the common rest.Config means every store automatically gets the auth behavior, if the Context contains an auth hint. Since every store currently uses the common rest.Config, I don't have to touch anything else. I could be more specific about bearer tokens (currently this just copies any Authorization header) I'll note that for future.

File: pkg/domains/k8s/client.go (hunk 2 above)
Comment (alanconway) [reply pkg/domains/k8s/client.go]: I took this back out, it was needless complexity. Ive made auth-copy mode "always-on" for now, we can refine later. Since korrel8r as a REST service always acts on behalf of the caller, I can't currently see any reason for it to use its own credentials, and several security risks if it did.

File: cmd/korrel8r/main.go (hunk 1 above)
Comment (xperimental) [reply cmd/korrel8r/main.go]: We might have talked past each other. This still uses `rest.Config` as before, just removes all the authentication methods from it (which might not just be bearer-token) and sets the specific field for bearer-token authentication in the client. This removes the need for a custom `RoundTripper` implementation but the "one `rest.Config` for all stores" is still the same.
